### PR TITLE
add --enable-seccomp

### LINKED ISSUE #119: add seccomp to sandbox

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Signed-off-by: Akihiro Suda <akihiro.suda.cz@hco.ntt.co.jp>

Fix #119

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
cannot be built for centos7 static
```console
$ docker build --target buildtest-centos76-static -f Dockerfile.buildtests .
...
#11 12.88 gcc -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include   -DCOMMIT="\"unknown\"" -g -O2  -static -o slirp4netns slirp4netns-main.o slirp4netns-slirp4netns.o slirp4netns-api.o slirp4netns-sandbox.o libslirp.a libparson.a -lglib-2.0   -lseccomp   -lpthread
#11 12.92 /usr/bin/ld: cannot find -lseccomp
...
```

File: Dockerfile.buildtests [hunk 1]
```diff
@@ -30,6 +30,11 @@ RUN ./configure && make && cp -f slirp4netns /
 
 FROM buildtest-centos76-common AS buildtest-centos76-static
 RUN yum install -y glibc-static glib2-static
+# FIXME: not sure how to install libseccomp-static via yum https://blog.51cto.com/bingdian/2347093
```
Comment (AkihiroSuda) [inline Dockerfile.buildtests]: @giuseppe @rhatdan PTAL? Is there any reason to exclude libseccomp-static from yum repo?

Comment (giuseppe) [reply Dockerfile.buildtests]: you could probably use it from the `https://cbs.centos.org/repos/virt7-container-common-candidate/x86_64/os/` repository:

```
# yum-config-manager --add-repo=https://cbs.centos.org/repos/virt7-container-common-candidate/x86_64/os/ && \
 yum install --nogpgcheck -y libseccomp-static
```
```

**Comment by giuseppe** [comment]
very nice feature!  LGTM

**Comment by rhatdan** [comment]
Is there any concern on performance?  Should we allow the user to be able to disable this feature?  I know that enabling seccomp has caused some performance issues.

File: Dockerfile.buildtests (hunk 1 above)
Comment (AkihiroSuda) [reply Dockerfile.buildtests]: thanks

**Comment by AkihiroSuda** [comment]
added as a new flag `--enable-seccomp`

File: zeccomp.h [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+/*
+ * the file name is "zeccomp.h" so as to avoid conflict with <seccomp.h> 
+ *
+ * FIXME: rename the file to "seccomp.h", and resolve the path conflict issue.
+ * It seems automake is adding -I. to CFLAGS.
+ */
```
Comment (AkihiroSuda) [inline zeccomp.h]: @giuseppe PTAL?

**Comment by AkihiroSuda** [comment]
rebased

File: zeccomp.h (hunk 2 above)
Comment (giuseppe) [reply zeccomp.h]: does the following patch work for you?

```
diff --git a/Makefile.am b/Makefile.am
index f882f8b..f94f9df 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,5 +1,6 @@
 bin_PROGRAMS = slirp4netns
 
+DEFAULT_INCLUDES =
 AM_CFLAGS = @GLIB_CFLAGS@
 
 noinst_LIBRARIES = libslirp.a libparson.a
diff --git a/slirp4netns.c b/slirp4netns.c
index de977a6..8c2770b 100644
--- a/slirp4netns.c
+++ b/slirp4netns.c
@@ -11,7 +11,7 @@
 #include "vendor/libslirp/src/libslirp.h"
 #include "api.h"
 #include "sandbox.h"
-#include "zeccomp.h"
+#include <seccomp.h>
 #include "slirp4netns.h"
 
 /* opaque for SlirpCb */
```

**Comment by giuseppe** [comment]
LGTM, the rename could be done as a followup PR

**Comment by rhatdan** [comment]
LGTM

File: zeccomp.h (hunk 2 above)
Comment (AkihiroSuda) [reply zeccomp.h]: thanks

Comment (AkihiroSuda) [reply zeccomp.h]: `make distcheck` fails 

```
make[3]: Entering directory '/slirp4netns/slirp4netns-0.4.0-beta.3+dev/_build/sub'
gcc -DHAVE_CONFIG_H     -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -DCOMMIT="\"unknown\"" -g -O2 -MT slirp4netns-main.o -MD -MP -MF .deps/slirp4netns-main.Tpo -c -o slirp4netns-main.o `test -f 'main.c' || echo '../../'`main.c
gcc -DHAVE_CONFIG_H     -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -DCOMMIT="\"unknown\"" -g -O2 -MT slirp4netns-slirp4netns.o -MD -MP -MF .deps/slirp4netns-slirp4netns.Tpo -c -o slirp4netns-slirp4netns.o `test -f 'slirp4netns.c' || echo '../../'`slirp4netns.c
Makefile:1375: recipe for target 'slirp4netns-main.o' failed
../../main.c:3:10: fatal error: config.h: No such file or directory
 #include "config.h"
          ^~~~~~~~~~
compilation terminated.
make[3]: *** [slirp4netns-main.o] Error 1
```

File: seccomp.c [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+/* FIXME: We should include <seccomp.h>, not </usr/include/seccomp.h>
+ *
+ * But including <seccomp.h> conflicts with "seccomp.h" in the current
+ * directory because Automake appends -I. to CFLAGS.
+ *
+ * We can eliminate -I. by setting DEFAULT_INCLUDES to empty value in
+ * Makefile.am, but that breaks `make distcheck`...
+ * https://github.com/rootless-containers/slirp4netns/pull/134#discussion_r316954960
+ */
+#include </usr/include/seccomp.h>
```
Comment (AkihiroSuda) [inline seccomp.c]: another workaround :(

**Pull request merged**